src/xdsl/util.py

```diff
@@ -35,6 +35,15 @@ def verify_(self) -> None:
     return TypeBase
 
 
+def fetch_metadata(f):
```

Comment:
Can't find the usage of this function, is it necessary?

Reply:
I think it's safe to remove